tornadoes/__init__.py

```diff
@@ -10,62 +10,66 @@
 
 
 class ESConnection(object):
+    _MATCH_ALL_QUERY = {"query": {"match_all": {}}}
 
-    def __init__(self, host='localhost', port='9200', io_loop=None, protocol='http', custom_client=None):
+    # TODO : timeout, max_retries, retry_on_timeout
+    def __init__(self, host='localhost', port='9200', io_loop=None, protocol='http', custom_client=None,
+                 http_request_kwargs=None):
         self.io_loop = io_loop or IOLoop.instance()
         self.url = "%(protocol)s://%(host)s:%(port)s" % {"protocol": protocol, "host": host, "port": port}
         self.bulk = BulkList()
         self.client = custom_client or AsyncHTTPClient(self.io_loop)
-        self.httprequest_kwargs = {}     # extra kwargs passed to tornado's HTTPRequest class e.g. request_timeout
+
+        # extra kwargs passed to tornado's HTTPRequest class e.g. request_timeout
+        self.http_request_kwargs = http_request_kwargs or {}
 
     @staticmethod
-    def from_uri(uri, io_loop=None, custom_client=None):
+    def _create_query_string(params):
+        """
+        Support Elasticsearch 5.X
+        """
+        parameters = params or {}
+
+        for param, value in parameters.items():
+            param_value = str(value).lower() if isinstance(value, bool) else value
+            parameters[param] = param_value
+
+        return urlencode(parameters)
+
+    @classmethod
+    def from_uri(cls, uri, io_loop=None, custom_client=None, http_request_kwargs=None):
         parsed = urlparse(uri)
 
         if not parsed.hostname or not parsed.scheme:
             raise ValueError('Invalid URI')
 
-        return ESConnection(**{
-            'host': parsed.hostname,
-            'protocol': parsed.scheme,
-            'port': parsed.port,
-            'io_loop': io_loop,
-            'custom_client': custom_client
-        })
+        return cls(host=parsed.hostname, protocol=parsed.scheme, port=parsed.port, io_loop=io_loop,
+                   custom_client=custom_client, http_request_kwargs=http_request_kwargs)
 
-    def create_path(self, method, **kwargs):
-        index = kwargs.pop('index', '_all')
-        doc_type = '/%s' % kwargs.pop('type', '')
+    @staticmethod
+    def create_path(method, index='_all', type='', **kwargs):
+        query_string = ESConnection._create_query_string(kwargs)
 
-        parameters = {}
-        for param, value in kwargs.items():
-            parameters[param] = value
+        path = "/%(index)s/%(type)s/_%(method)s" % {"method": method, "index": index, "type": type}
```

Comment:
you understand that's not my code right?
There are a lot more things I would change in this code but I changed almost only what I must changed.
But anyway, I don't think it's so bad the way they wrote it, I would prefer it by your way.